runtime/tests/interpreter/values_test.go

```diff
@@ -359,6 +359,75 @@ func TestInterpretRandomMapOperations(t *testing.T) {
 		assert.Equal(t, startingSlabCounts, slabCounts)
 	})
 
+	t.Run("update enum key", func(t *testing.T) {
+
+		dictionary := interpreter.NewDictionaryValueWithAddress(
+			inter,
+			interpreter.EmptyLocationRange,
+			&interpreter.DictionaryStaticType{
+				KeyType:   interpreter.PrimitiveStaticTypeAnyStruct,
+				ValueType: interpreter.PrimitiveStaticTypeAnyStruct,
+			},
+			orgOwner,
+		)
+
+		require.Equal(t, 0, dictionary.Count())
+
+		newEntries := newValueMap(numberOfValues)
+
+		value1 := interpreter.NewUnmeteredIntValueFromInt64(1)
+		value2 := interpreter.NewUnmeteredIntValueFromInt64(2)
+
+		keyValues := make([][2]interpreter.Value, numberOfValues)
+		for i := 0; i < numberOfValues; i++ {
+			// Create a random enum as key
+			key := r.generateRandomHashableValue(inter, randomValueKindEnum)
+
+			newEntries.put(inter, key, value1)
+
+			keyValues[i][0] = key
+			keyValues[i][1] = value1
+		}
+
+		// Insert
+		for _, keyValue := range keyValues {
+			dictionary.Insert(inter, interpreter.EmptyLocationRange, keyValue[0], keyValue[1])
+		}
+
+		// Update
+		newEntries.foreach(func(orgKey, orgValue interpreter.Value) (exit bool) {
+			oldValue := dictionary.Insert(
+				inter,
+				interpreter.EmptyLocationRange,
+				orgKey.Clone(inter),
+				// change value1 to value2
+				value2,
+			)
+
+			require.IsType(t, &interpreter.SomeValue{}, oldValue)
+			someValue := oldValue.(*interpreter.SomeValue)
+
+			// Removed value must be same as the original value
+			innerValue := someValue.InnerValue(inter, interpreter.EmptyLocationRange)
+			utils.AssertValuesEqual(t, inter, orgValue, innerValue)
+
+			return false
+		})
+
+		// Check the values
+		newEntries.foreach(func(key, _ interpreter.Value) (exit bool) {
```

Comment:
Does the second parameter need to be named `value` so it can be compared with `innerValue` later?

```suggestion
		newEntries.foreach(func(key, value interpreter.Value) (exit bool) {
```

Reply:
I improved the test case a bit more in [96ac399](https://github.com/onflow/cadence/pull/3556/commits/96ac39915cb8b99fa4fc519dbd87dc1fd14b868f)